lib/dc/store/aws_s3_store.rb

```diff
@@ -14,13 +14,14 @@ module AwsS3Store
 
       # 60 seconds for persistent connections.
       S3_PARAMS       = {:connection_lifetime => 60}
+      AWS_REGION      = DC::CONFIG['aws_region']
 
       ACCESS_TO_ACL   = Hash.new(:private)
       DC::Access::PUBLIC_LEVELS.each{ |level| ACCESS_TO_ACL[level] = :public_read }
 
       module ClassMethods
         def asset_root
-          "https://s3.amazonaws.com/#{BUCKET_NAME}"
+          "https://s3-#{AWS_REGION}.amazonaws.com/#{BUCKET_NAME}"
```

Comment:
@reefdog Damn! `us-east-1` is the only exception to the rule. An if statement should solve it.

Ref: http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region